src/init/starship.ps1

```diff
@@ -1,16 +1,50 @@
 #!/usr/bin/env pwsh
 
+function Get-Cwd {
+    $cwd = Get-Location
+    $provider_prefix = "$($cwd.Provider.ModuleName)\$($cwd.Provider.Name)::"
+    return @{
+        # Resolve the actual/physical path
+        # NOTE: ProviderPath is only a physical filesystem path for the "FileSystem" provider
+        # E.g. `Dev:\` -> `C:\Users\Joe Bloggs\Dev\`
+        Path = $cwd.ProviderPath;
+        # Resolve the provider-logical path 
+        # NOTE: Attempt to trim any "provider prefix" from the path string.
+        # E.g. `Microsoft.PowerShell.Core\FileSystem::Dev:\` -> `Dev:\`
+        LogicalPath =
+            if ($cwd.Path.StartsWith($provider_prefix)) {
+                $cwd.Path.Substring($provider_prefix.Length)
+            } else {
+                $cwd.Path
+            };
+    }
+}
+
+function Fix-Arg ($arg) {
```

Comment:
Manual quoting doesn't work? And backslashes aren't even supposed to be escape characters in powershell. Also why would the shell need to add literal quotes?

Reply:
Edit: looks like this is an actual issue but you could use `.TrimEnd("\")`

Reply:
This isn't an issue with escaping characters in PowerShell code, but rather in the way PS generates the [Windows command-line argument string](https://docs.microsoft.com/en-us/previous-versions//17w5ykft(v=vs.85)?redirectedfrom=MSDN) when it invokes an external program, and there seems to be a bug in this routine.

In the documentation I linked above, note the final three rules:
- Backslashes are interpreted literally, unless they immediately precede a double quotation mark.
- If an even number of backslashes is followed by a double quotation mark, one backslash is placed in the argv array for every pair of backslashes, and the double quotation mark is interpreted as a string delimiter.
- If an odd number of backslashes is followed by a double quotation mark, one backslash is placed in the argv array for every pair of backslashes, and the double quotation mark is "escaped" by the remaining backslash, causing a literal double quotation mark (") to be placed in argv.

E.g. for the following external command invocation:
```powershell
> $a = 'hello "world" goodbye "moon"'
> $b = 'path\without\whitespace\'
> $c = 'path\with whitespace\'
> externalprogram $a $b $c
```
PS generates the following Windows command-line string:
```cmd
externalprogram "hello \"world\" goodbye \"moon\"" path\without\whitespace\ "path\with whitespace\"
```

The first argument is quoted correctly including escaped quote marks. ✔️ 
The second argument is not quoted at all as it does not contain any whitespace. ✔️ 
The third argument however has whitespace _and_ a trailing, odd number of backslashes which ends up escaping the closing quote mark. ❌

PowerShell could handle this by detecting a trailing backslash and escaping it as part of the quoted command-line string serialisation, but for some reason it doesn't. I'm not sure why not - it seems an obvious oversight.
I'm using PowerShell 5.0 - I haven't tested with the cross-platform PowerShell version yet so maybe they've fixed it there.

Reply:
In reading through that PowerShell issue thread I learnt about the the [stop-parsing symbol (`--%`)](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_parsing?view=powershell-7.1). This was introduced in PowerShell 3.0 (released in 2012) so I think we can rely on it being available.

With the stop-parsing symbol, we can manually construct a command string which we know will be correctly parsed and PS will pass it unchanged to the Windows API.

Reply:
I've been experimenting with an `Invoke-Native` function over here:
https://github.com/deadalusai/printargs/blob/master/test.ps1

This function works around the argument escaping issue by quoting/escaping arguments manually and using `--%` to pass the argument string through to the rust executable directly. The string is quoted using the Windows command-line argument string rules, but the code _appears_ to work on Linux (Ubuntu/Mint) correctly too.

Reply:
Ok, it works correctly on Linux/Unix because the .NET runtime on Linux itself [parses the argument string into the `argv` collection](https://github.com/dotnet/runtime/blob/9c41693194e3b005c91ee471ab3e649d2479d34d/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs#L430) before forking the new process, so we don't need to rely on distro-specific argv parsing behavior.

["This follows the rules outlined in "Parsing C++ Command-Line Arguments" at https://msdn.microsoft.com/en-us/library/17w5ykft.aspx."](https://github.com/dotnet/runtime/blob/9c41693194e3b005c91ee471ab3e649d2479d34d/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs#L786-L789).

Reply:
I've updated the PR with a more robust solution for invoking native executables (`Invoke-Native`) which ensures that the strings passed as arguments make their way into the Rust argv list in the same shape as they started, trailing backslashes and all.

There is also a PowerShell library called [Native](https://github.com/mklement0/Native) available on PSGallery which supports all platforms and works by forking out to `/bin/bash` or `cmd`, but that would be an external dependency which may not be desirable.